### PR TITLE
BIG-242 region from genericbackendcredentials

### PR DESCRIPTION
**Pull request by martinjandl**
Jira: BIG-242
Connection PR: [4920](https://github.com/keboola/connection/pull/4920)
SAPI PR: XXX

---

### CODE REVIEW

**Comment by romanbracinik** [comment]
👁️

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ Create a sub folder in the **KBC Team Dev** (id: [431160969986](https://console.
 1. get missing pieces (organization_id and billing_id) from [Connection repository](https://github.com/keboola/connection/blob/master/docs/DOCKER.md#bigquery).
 2. (optional) move `bq-storage-backend-init.tf` out of project directory so new files would be out of git
 3. Run `terraform init` 
-4. Run `terraform apply -var folder_id=[folder_id] -var billing_account_id=[billing_id] -var backend_prefix=<your prefix, eg. js-driver-bq> -var region=<desired region>`
+4. Run `terraform apply -var folder_id=[folder_id] -var billing_account_id=[billing_id] -var backend_prefix=<your prefix, eg. kbc-js> -var region=<desired region>`
```
Comment (romanbracinik) [inline README.md]: ❤️

File: README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ Create a sub folder in the **KBC Team Dev** (id: [431160969986](https://console.
 1. get missing pieces (organization_id and billing_id) from [Connection repository](https://github.com/keboola/connection/blob/master/docs/DOCKER.md#bigquery).
 2. (optional) move `bq-storage-backend-init.tf` out of project directory so new files would be out of git
 3. Run `terraform init` 
-4. Run `terraform apply -var folder_id=[folder_id] -var billing_account_id=[billing_id] -var backend_prefix=<your prefix, eg. kbc-js> -var region=<desired region>`
+4. Run `terraform apply -var folder_id=[folder_id] -var billing_account_id=[billing_id] -var backend_prefix=<your prefix, eg. kbc-js> -var file_storage_backend_region=<desired region>` (e.g. us-central1 region)
```
Comment (romanbracinik) [inline README.md:21]: ❤️

File: README.md [hunk 3]
```diff
@@ -7,6 +7,12 @@ Keboola high level storage backend driver for Big Query
 Install [Google Cloud client](https://cloud.google.com/sdk/docs/install-sdk) (via [Brew](https://formulae.brew.sh/cask/google-cloud-sdk#default)), initialize it
 and log in to [generate default credentials](https://cloud.google.com/docs/authentication/application-default-credentials#personal).
 
+
+Install dependencies.
+```bash
+docker-compose run --rm dev composer install
+```
+
```
Comment (romanbracinik) [inline README.md]: Super ze si to pridal, ale premyslam ci by sa tam nemalo radsej presunut 
```
## Build docker images

bash
docker-compose build
```

sa ti ta rovno naistaluje ten composer

File: README.md [hunk 4]
```diff
@@ -45,6 +45,9 @@ BQ_KEY_FILE=<big_query_key.json file owned by main service acc>
 ```
 
 All done. Now you can try `composer loadGcs` script and run tests.
+```bash
```
Comment (romanbracinik) [inline README.md:48]: 👍

File: src/CredentialsMetaRequiredException.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+class CredentialsMetaRequiredException extends Exception implements NonRetryableExceptionInterface
+{
+    public function __construct(
+        string $message,
+        int $code = self::ERR_VALIDATION,
+        ?Throwable $previous = null,
+    ) {
+        parent::__construct(
+            $message,
```
Comment (romanbracinik) [inline src/CredentialsMetaRequiredException.php]: Len drobnost, asi by som tu message hodil do vnutra tej exception

**Review by romanbracinik — APPROVED** [review]
lgtm, nezabudni potom odstanit este tie uz nepouzivane parametre z https://github.com/keboola/storage-backend/tree/main/packages/php-storage-driver-common/proto a requirnut novu verziu tu v driveru

File: README.md (hunk 3 above)
Comment (martinjandl) [reply README.md]: Pro ten build uz by mely byt nastavene envy.. a composer install byl myslim potreba pred composer loadGcs

File: src/CredentialsMetaRequiredException.php (hunk 5 above)
Comment (martinjandl) [reply src/CredentialsMetaRequiredException.php]: upravil jsem

**Comment by martinjandl** [comment]
Upraveny php-storage-driver-common
https://github.com/keboola/storage-backend/pull/138

**Review by romanbracinik — APPROVED** [review]
🙆‍♂️

**Comment by martinjandl** [comment]
Nalinkoval jsem keboola/storage-driver-common 6.8

**Review by romanbracinik — APPROVED** [review]
lgtm

**Pull request merged**